src/components/NavBar/ChainSelector.tsx

```diff
@@ -72,6 +70,9 @@ export const ChainSelector = ({ leftAlign }: ChainSelectorProps) => {
 
   const theme = useTheme()
 
+  const showTestnets = useAtomValue(showTestnetsAtom)
+  const chains = NETWORK_SELECTOR_CHAINS.filter((chain) => !(chain === SupportedChainId.SEPOLIA && !showTestnets))
```

Comment:
For future testnet addition can you add a `TESTNETS` array in `src/constants/chains.ts` which contains all testnets and then here you can just check if chain is in that array